### ISSUE TITLE
integrate error "not of type FIXNUM" for integrand with floats in it

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1324]
Imported from SourceForge on 2024-07-04 09:49:04
Created by **[zmth](https://sourceforge.net/u/zmth/)** on 2021-07-10 00:28:44
Original: https://sourceforge.net/p/maxima/bugs/3810

---

build_info()  or bug_report()
"branch_5_44_base_231_g5c411f69f",timestamp="2021-01-12 23:51:42",host="x86_64-w64-mingw32",lisp_name="SBCL",lisp_version="2.0.0",maxima_userdir="C:/Users/zmth1/maxima",maxima_tempdir="C:/Users/zmth1/AppData/Local/Temp",maxima_objdir="C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0",maxima_frontend="wxMaxima",maxima_frontend_version="20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks")
~~~
(assume_pos:true, file_output_append:true, ratprint:false, showtime:true,load(simplify_sum)
, simpsum:true,  load("lrats"),letrat:true,ratfac:true,algebraic:true,factor_max_degree_print_warning:false
,kill(nn),init(nn):=block(lisf:makelist(1,4*nn+5),lisf[3]:2,for i:3 thru 4*nn+4 do lisf[i+1]:i*lisf[i]
, fg:makelist(1,6+4*nn),fg[2]:1/2,for i:2 thru 4*nn+2 do fg[i+1]:/*float*/((2*i-1)*fg[i]/2),matffule:zeromatrix
(5*nn+1,10*nn+2),for j thru 10*nn+2 do(matffule[1,j]:1,if j>1 then matffule[2,j]:j-1
,for i:3 thru min(5*nn+1,j/2+1) do matffule[i,j]:matffule[i-1,j]*(j+3-2*i))
, matjL:zeromatrix(2*nn+2,2*nn+2),for j thru nn+1 do(matjL[1,j]:1,matjL[2,j]:j,for i:3 thru 
max(2*(nn-j+1),nn-1)+2 do matjL[i,j]:matjL[i-1,j]*(j+i-2)),for j:nn+2 thru 2*nn+1 do
(matjL[1,j]:1,matjL[2,j]:j,for i:3 thru nn+1 do matjL[i,j]:matjL[i-1,j]*(j+i-2))
,matjL[1,2*nn+2]:1,matj3:zeromatrix(nn+1,nn+1),for j thru nn +1 do(matj3[1,j]:1
,if j>1 then matj3[2,j]:j-1/2,for i:3 thru min(j,nn+1) do matj3[i,j]:matj3[i-1,j]*(j-i+3/2))
,binom:zeromatrix(2*nn+1,2*nn+1),for i:0 thru 2*nn do for j:0 thru i do binom[i+1,j+1]
:matjL[i-max(j,i-j)+1,max(j,i-j)+1]/matjL[i-max(j,i-j)+1,1])/*end blk*/,init(12),fpprintrec:6

    ,kill(x),fl:true,Lgndre(L1,w):= 2^(-L1-1/2)*sqrt(2*L1+1)*sum((-1)^k1*binomial(L1,k1)*binomial(2*(L1-k1),L1)*w^(L1-2*k1),k1
,0,L1/2), jpoly(j,L,x):=(-1)^j*2^(j+L+1)*sqrt(2*(L+j+1)*prod(j+i,i,1,2*L+1)/%pi)*prod(2*L+2*j-2*i,i,0,j/2-1)/prod(2*L+2*floor
(j/2)+1-2*i,i,0,(2*L+j-1)/2)*sum((-1)^k*prod(j+L+3/2-i,i,1,j-k)*prod(i,i,k+1,j)*x^k/prod(2*(j+L)+1-i,i,0,j-k-1)/(j-k)!,k,0,j)

,hyp(cnuc,lisn,re):=block(flx:true,ind1:makelist(0,(nn+1)*(nn**2+5*nn+6)/6)
,for L:0 thru nn do (itL:nn-L+1, iLn:L*(2*nn-L+3)/2,iL:(L*(3*nn*(nn-L)+12*nn+L**2-6*L+11))/6
,mx:ident(itL),matmp:zeromatrix(itL,itL),for n1 thru itL do(k2:rlis[iLn+n1] ,n:lisn[iLn+n1],mx[n1,n1]:float(lisf[n+4]*(2*(n+4)
*(n+5)+cnuc*k2**2*(n+16*L**2+32*L+20))/2^(n+7)/k2^(n+6)),for m1 thru n1-1 do(k12:k2+(rlis[iLn+m1]),iiL:lisn[iLn+m1]+n+8
, ss:float(((iiL+2)*2*iiL+cnuc*(k12*(k12*(16*L*(L+2)-n*(n+8))+2*k2*iiL*(n+5))-iiL*(iiL+2)*k2^2))/8/k12**(floor((iiL+4)/2)))

,if mod(iiL,2)=0 then ss:float(ss*lisf[iiL/2]) else ss:float(ss*sqrt(%pi/k12)*fg[(iiL+1)/2]),mx[n1,m1]:ss, mx[m1,n1]:ss) )
,ind1[iL+1]:float(1/sqrt(mx[1,1])),matmp[1,1]:ind1[iL+1],for j:2 thru itL do(nj:j*(j-1)/2,ind1[iL+nj+j]:1,s2:mx[j,j],for i thru j-1 do
(ni:i*(i-1)/2,ti:float(sum(ind1[iL+ni+k]*mx[k,j],k,1,i)),for k thru i do ind1[iL+nj+k]:ind1[iL+nj+k]-ti*ind1[iL+ni+k],s2:s2-ti^2 )
,s2r:s2/mx[j,j],if s2r<emin then(emin:s2r,Lmin:L,if s2r>emax then emax:s2r, if s2r<.0004 then(to:["nn",nn,"L",L,"j",j,"s2r too small"
,s2r,"re",re],disp(to),write_data(to,"indx.t"),flx:false)), if flx then(s2:float(1/sqrt(s2)),for i thru j do ind1[iL+nj+i]:float
(s2*ind1[iL+nj+i]), for i thru j do matmp[i,j]:ind1[iL+nj+i]))/*end j*/ ,if flx then(matmp:transpose(matmp).mx.matmp,for i thru itL 
do matmp[i,i]:matmp[i,i]-1  ,lmo:float(abs(list_matrix_entries(matmp))),rm:lmax(lmo),if rm>roff then(roff:rm,if rm>.001 then(flx:false
,to:["nn",nn,"hyp,roff>.01, L",L,"roff",roff,"re",re],disp(to),write_data(to,"indx.t") ))))/*end L ,finished ke+1*/      
,lisrbk:makelist(0,(nn+1)*(nn+2)*(nn**2+3*nn+4)/8),INT:makelist(0,(nn+1)*(nn+2)*(nn**2+3*nn+4)/8)

,cnt:0,for L1:0 thru nn while flx do(iL1: (L1*(3*nn*(nn-L1)+12*nn+L1**2-6*L1+11))/6,itl1:nn-L1+1,iLn1:L1*(2*nn-L1+3)/2
,for n1 thru itl1 do(in1:iL1+n1*(n1-1)/2,S2a:sum(ind1[in1+i]*x^(lisn[iLn1+i]/2)*exp(-rlis[iLn1+i]*x),i,1,n1),for L2:0 thru 
L1 do(iL2:(L2*(3*nn*(nn-L2)+12*nn+L2**2-6*L2+11))/6 ,itl2:nn-L2+1,iLn2:L2*(2*nn-L2+3)/2,if L2#L1 then iu:itl2 else iu:n1
,for n2 thru iu do(in2:iL2+n2*(n2-1)/2,S2b:sum(ind1[in2+j]*x^(lisn[iLn2+j]/2)*exp(-rlis[iLn2+j]*x),j,1,n2),cnt:cnt+1,s2:0
,for i thru n1 do for j thru n2 do (rk12:rlis[iLn1+i]+rlis[iLn2+j],iiL:lisn[iLn1+i]+lisn[iLn2+j]+8 ,si:float(ind1[in1+i]
*ind1[in2+j]/rk12^(floor(iiL/2)+1)),if mod(iiL,2)=0 then s2:s2+float(si*lisf[iiL/2+1]) else s2:s2+float(si*fg[(iiL+3)/2]
*sqrt(%pi/rk12))) ,lisrbk[cnt]:s2,disp([S2a*S2b]),INT[cnt]:float(integrate(x^4*S2a*S2b,x,0,inf)),if abs(lisrbk[cnt]-INT[cnt])>1e-11 then
(fl:false,disp(["L1",L1,"n1",n1,"L2",L2,"n2",n2,"cnt",cnt,"INT",INT[cnt],"lisrbk",lisrbk[cnt]]))))))))$

(nn:12,nrec:nn,cf:-1,cn:2,neo:-3,nn1:(nn+1)*(nn+2)/2,lisn:makelist(0,nn1),rlis:makelist(0,nn1),rkmn:sqrt(2)
,rkmx:rkmn*(1+nn/22),re:float(1.06),ne:4*nn+neo,emin:9,roff:0,itt:(6*floor((nn+2)/2)+(nn+1)*(nn+2)*(2*nn+9))/24
,for L:0 thru nn do (itL:nn-L+1,rk:(rkmx*(itL-1)+L*rkmn)/nn,lisn[(L+1)*(2*nn-L+2)/2]:ne, iLn:L*(2*nn-L+3)/2
,for i thru nn-L do(lisn[iLn+i]:4*(L+i-1)+neo,rlis[iLn+i]:float(re^(ne-lisn[iLn+i])*rk)) ,rlis[iLn+itL]:float(re^
(ne-lisn[iLn+itL])*rk)),it1:itt*(itt+1)/2,cnuc:1,hyp(cnuc,lisn,re));
~~~

it  ignores fpprintprec:6. Also unrelated problem of get message 
 The value
   -27613027758756006981857458587882876082753548456102059144159
 is not of type   FIXNUM 
 that happens regardless even if take out all disp(...)  commands . Also unless one wants to wait practically forever ctrl G  interrupt or anythng is useless,

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 09:49:06
Created by **[zmth](https://sourceforge.net/u/zmth/)** on 2021-07-10 02:34:08
Original: https://sourceforge.net/p/maxima/bugs/3810/#52b0

---

note is it is very inconsistent. For no user reason at all sometimes it does not give the error:
 The value
xxxxxx...
 is not of type   FIXNUM 
 and sometimes it does under the exact same user conditions. It even once gave for only nn:9
 The value -148680189678384495934 is not of type   FIXNUM
 
 If u can't get it to be inconsistent on your end i will put back in the full program and then check it.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 09:49:10
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-07-14 07:12:40
Original: https://sourceforge.net/p/maxima/bugs/3810/#ea1f

---

The name of the global variable is misspelled, that's why it has no effect -- you want `fpprintprec`, not `fpprintrec`.

About the "not of type FIXNUM", this appears to be a consequence of floats appearing in the integrand. Maxima is trying to factor a polynomial which has coefficients created by rationalizing floats, and it ends up with rationals with enormous numerator or denominator. I will look at trying to avoid the error in the polynomial code, but a workaround would be to avoid floats in the integrand. Try to do as much as you can without floats, and then convert to floats only at the very end.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 09:49:13
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-07-14 17:03:31
Original: https://sourceforge.net/p/maxima/bugs/3810/#b7dd

---

- **labels**:  --> integrate, floating point, polynomials
- **summary**: ignores fpprintprec etc. --> integrate error &#34;not of type FIXNUM&#34; for integrand with floats in it
- **Group**: Includes_proposed_fix --> None

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 09:49:20
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-09-16 06:42:08
Original: https://sourceforge.net/p/maxima/bugs/3810/#5510

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-04 09:49:24
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2021-09-16 06:42:08
Original: https://sourceforge.net/p/maxima/bugs/3810/#efe7

---

Fixed by commit a5622ba.
